src/package_index.rs

```diff
@@ -18,6 +18,7 @@ use crate::package_name::PackageName;
 #[derive(Debug, Serialize, Deserialize)]
 pub struct PackageIndexConfig {
     pub api: Url,
+    pub oauth_id: Option<String>,
```

Comment:
We might have other auth modes in the future. Do you think that `github_oauth_id` would be a good name?

Reply:
Yeah that seems reasonable!